sklearn/decomposition/_dict_learning.py

```diff
@@ -1706,24 +1753,30 @@ def fit(self, X, y=None):
         """
         self._validate_params()
 
+        _check_positive_coding(method=self.fit_algorithm, positive=self.positive_code)
+
+        method = "lasso_" + self.fit_algorithm
+
         random_state = check_random_state(self.random_state)
+
```

Comment:
you can revert this change

Reply:
You mean these conditions should be moved back into _dict_learning? 
Also should we check random state here as well as in the original method or once is enough?

Reply:
I meant the new line. We tend to not create diff that is not necessary.